src/main/java/com/esri/core/geometry/ogc/OGCConcreteGeometryCollection.java

```diff
@@ -646,93 +649,102 @@ public boolean Equals(OGCGeometry another) {
 		
 		OGCConcreteGeometryCollection gc1 = (OGCConcreteGeometryCollection)g1;
 		OGCConcreteGeometryCollection gc2 = (OGCConcreteGeometryCollection)g2;
+		// TODO Assuming input geometries are simple and valid, remove-overlaps would be a no-op.
+		// Hence, calling flatten() should be sufficient.
 		gc1 = gc1.flattenAndRemoveOverlaps();
 		gc2 = gc2.flattenAndRemoveOverlaps();
 		int n = gc1.numGeometries();
 		if (n != gc2.numGeometries()) {
 			return false;
 		}
 		
-		boolean res = false;
 		for (int i = 0; i < n; ++i) {
 			if (!gc1.geometryN(i).Equals(gc2.geometryN(i))) {
 				return false;
 			}
-			
-			res = true;
 		}
 		
-		return res;
+		return n > 0;
 	}
-	
+
+	private static OGCConcreteGeometryCollection toGeometryCollection(OGCGeometry geometry)
+	{
+		if (geometry.geometryType() != OGCConcreteGeometryCollection.TYPE) {
+			return new OGCConcreteGeometryCollection(geometry, geometry.getEsriSpatialReference());
+		}
+
+		return (OGCConcreteGeometryCollection) geometry;
+	}
+
+	private static List<Geometry> toList(GeometryCursor cursor)
+	{
+		List<Geometry> geometries = new ArrayList<Geometry>();
+		for (Geometry geometry = cursor.next(); geometry != null; geometry = cursor.next()) {
+			geometries.add(geometry);
+		}
+		return geometries;
+	}
+
 	//Topological
 	@Override
 	public OGCGeometry difference(OGCGeometry another) {
-		List<OGCConcreteGeometryCollection> list = wrapGeomsIntoList_(this, another);
-		list = prepare_for_ops_(list);
```

Comment:
Ok. Let's have it simpler. The results will be slightly different than if (or when) the operation would be implemented in the topograph directly, the preparation step you removed was there to make sure it would not change.